crates/cast/bin/cmd/run.rs

```diff
@@ -217,6 +230,27 @@ impl RunArgs {
             }
         };
 
+        if self.generate_local_signatures {
+            let project = config.project()?;
+            let compiler = ProjectCompiler::new().quiet(true);
+            let out = compiler.compile(&project)?;
+            let mut cached_signatures = CachedSignatures::default();
+            out.artifacts().for_each(|(_, artifact)| {
+                if let Some(abi) = &artifact.abi {
+                    for func in abi.functions() {
+                        cached_signatures.functions.insert(func.selector().to_string(), func.signature());
+                    }
+                    for event in abi.events() {
+                        cached_signatures.events.insert(event.selector().to_string(), event.full_signature());
+                    }
+                }
+            });
+            let path = Config::foundry_cache_dir().unwrap().join("signatures");
+            if path.is_file() {
+                let _ = fs::write_json_file(&path, &cached_signatures);
+            }
+        }
+
         handle_traces(result, &config, chain, self.label, self.debug).await?;
```

Comment:
perhaps a better solution than doing the preprocess step would be to provide the local signatures here:

so they can be merged with this here

https://github.com/foundry-rs/foundry/blob/eef87de35375311f134fb4261c18455ced8022a5/crates/cli/src/utils/cmd.rs#L393-L396

Reply:
In fact, it makes more sense for this function to be executed at **forge compile** time, and it apears in **cast run** as a shortcut entry to avoid typing the command twice.